pkg/applicationprofilemanager/v1/applicationprofile_manager.go

```diff
@@ -495,6 +474,16 @@ func (am *ApplicationProfileManager) startApplicationProfiling(ctx context.Conte
 		NsMntId:          container.Mntns,
 	}
 
+	// don't start monitoring until we have the instanceID - need to retry until the Pod is updated with containerID
```

Comment:
This is tricky because we might lose data.
My suggestion is to execute this task in a separate process, even if it means accepting the risk of the container ending before we can retrieve the parent.